src/Workspaces/CSharp/Portable/SyntaxDifferencing/CSharpSyntaxDifferenceService.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.SyntaxDifferencing;
+
+namespace Microsoft.CodeAnalysis.CSharp.SyntaxDifferencing
```

Comment:
It would be better to keep this in EditAndContinue namespace and perhaps also include 'EditAndContinue' to the name of the service. The implementation of the syntax tree differencing hat's used here is very specific to EnC, especially on method body level. For example, we ignore all expression nodes and only pick lambda expression from them for the tree diffing alg. So ```F(1 + 2);``` changed to ```F(1 + 3);``` results in an edit of the entire statement, not in an edit of the literal expression as one might expect from general syntax diffing impl.